modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java

```diff
@@ -134,12 +163,23 @@ private CreateTableCommand convertCreateTable(IgniteSqlCreateTable createTblNode
         createTblCmd.schemaName(deriveSchemaName(createTblNode.name(), ctx));
         createTblCmd.tableName(deriveObjectName(createTblNode.name(), ctx, "tableName"));
         createTblCmd.ifTableExists(createTblNode.ifNotExists());
+        createTblCmd.dataStorage(driveDataStorage(createTblNode.engineName(), ctx));
 
         if (createTblNode.createOptionList() != null) {
-            for (SqlNode optNode : createTblNode.createOptionList().getList()) {
-                IgniteSqlCreateTableOption opt = (IgniteSqlCreateTableOption) optNode;
+            for (SqlNode optionNode : createTblNode.createOptionList().getList()) {
+                IgniteSqlCreateTableOption option = (IgniteSqlCreateTableOption) optionNode;
+
+                assert option.key().isSimple() : option.key();
+
+                String optionKey = option.key().getSimple();
 
-                tblOptionProcessors.getOrDefault(opt.key(), UNSUPPORTED_OPTION_PROCESSOR).process(opt, ctx, createTblCmd);
+                if (tableOptionInfos.containsKey(optionKey)) {
+                    processTableOption(tableOptionInfos.get(optionKey), option, ctx, createTblCmd);
+                } else if (dataStorageOptionInfos.get(createTblCmd.dataStorage()).containsKey(optionKey)) {
```

Comment:
Does it make sense to require an engine to be specified explicitly if any storage's option is provided?

Reply:
@agura Should we require to specify an engine if at least one property of any engine is specified?

Reply:
A storage engine should be infered from set of configuration, SET default_engine command and ENGINE clause. Any engine's property should be validated against available set of the inferred storage engine. There is no sense to infer a storage engine from properties defined in WITH clause because it is not transparent for an end user and different stroages can have the same property name.

Infer gets a storage engine using the following priority:

 - ENGINE clause - overrides all defaults, if defined
 - default_engine - overrides configuration default, if defined
 - defaultStroageEngine from configuration - in other cases (should always be defined)

Reply:
@agura, does it make sense to allow a syntax like this:
`CREATE TABLE my_t (id int primary key, val int) WITH specific_storage_engine_propkey=specific_storage_engine_propval`

This will render a different results depends on what is the actual value of `defaultStroageEngine` configuration property. In my opinion, it's more safe to require the engine to be specified either by ENGINE clause or  current execution context.

Reply:
@korlov42 @tkalkirill I propose to implement the behavior described above. We can change this behavior easy later.

Reply:
Ok